gprofiler/metadata/py_module_version.py

```diff
@@ -0,0 +1,282 @@
+    try:
+        f = open(f"/proc/{pid}/maps")
+    except OSError:
+        return None
+
+    with f:
+        for line in f.readlines():
+            match = _LIBPYTHON_MAPS_PATTERN.search(line.strip())
+            if match is not None:
+                return match.group()
+    return None
+
+
+# Matches PY_VERSION in Python's binary
+_PY_VERSION_STRING_PATTERN = re.compile(rb"(?<=\D)(?:2\.7|3\.1?\d)\.\d\d?(?=\x00)")
+
+
+@functools.lru_cache(maxsize=128)
+def _get_python_full_version(process: Process) -> Optional[str]:
+    bin_file = _get_libpython_path(process.pid) or f"/proc/{process.pid}/exe"
+
+    # Try to extract the version string from the binary
+    try:
+        f = open(resolve_host_path(process, bin_file), "rb")
+    except OSError:
+        return None
+
+    with f:
+        for line in f.readlines():
+            match = _PY_VERSION_STRING_PATTERN.search(line)
+            if match is not None:
+                return match.group().decode()
+    return None
+
+
+# Standard library modules are identified by being under a pythonx.y dir and *not* under site/dist-packages
+_STANDARD_LIB_PATTERN = re.compile(r"/python\d\.\d\d?/(?!.*(site|dist)-packages)")
+
+
+def _get_standard_libs_version(result: Dict[str, Optional[Tuple[str, str]]], process: Process):
+    py_version = None
+
+    for path in result:
+        match = _STANDARD_LIB_PATTERN.search(path)
+        if match is None:
+            # This module is (probably) not part of the standard library
+            continue
+
+        if py_version is None:
+            py_version = _get_python_full_version(process)
+            if py_version is None:
+                # No need to continue trying if we failed
+                return None
+        # (mypy fails to understand that py_version isn't Optional at this point)
+        result[path] = ("standard-library", py_version)  # type: ignore
+
+
+@functools.lru_cache(maxsize=128)
+def _get_packages_files(process: Process, packages_path: str) -> Dict[str, Tuple[str, str]]:
+    """Return a dict of filename: (package_name, package_version) for the packages in packages_path"""
+    # Transform packages_path to be relative to /proc/[pid]/root/
+    packages_host_path = resolve_host_path(process, packages_path)
+
+    path_to_package_info = {}
+    for dist in pkg_resources.find_distributions(packages_host_path):
+        files_iter = _files_from_record(dist) or _files_from_legacy(dist)
+        if files_iter is not None:
+            pacakge_name = _get_package_name(dist)
+            if pacakge_name is not None:
+                path_to_package_info.update(
+                    dict.fromkeys(
+                        (os.path.join(packages_path, file) for file in files_iter), (pacakge_name, dist.version)
+                    )
+                )
+    return path_to_package_info
+
+
+_warned_no__normalized_cached = False
+
+
+def _get_packages_versions(result: Dict[str, Optional[Tuple[str, str]]], process: Process):
```

Comment:
The name `result` is very confusing in this context. 
1. It'd be better named `packages_versions`.
2. The function would be better named `populate_packages_versions` to signify that it modifies the parameter and does not return a value.